hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -248,6 +249,24 @@ private[hudi] object HoodieSparkSqlWriter {
     }
   }
 
+  /**
+   * Checks if schema needs upgrade (if incoming records's schema is old while table schema got evolved).
+   * @param fs instance of FileSystem.
+   * @param basePath base path.
+   * @param sparkContext instance of spark context.
+   * @param schema incoming record's schema.
+   * @return Pair of(boolean, table schema), where first entry will be true only if schema conversion is required.
+   */
+  def getLatestSchema(fs: FileSystem, basePath: Path, sparkContext: SparkContext, schema: Schema) : Schema = {
+    var latestSchema: Schema = schema
+    if(fs.exists(new Path(basePath.toString + "/" + HoodieTableMetaClient.METAFOLDER_NAME))) {
+      val tableMetaClient = HoodieTableMetaClient.builder.setConf(sparkContext.hadoopConfiguration).setBasePath(basePath.toString).build()
+      val tableSchemaResolver = new TableSchemaResolver(tableMetaClient)
+      latestSchema = tableSchemaResolver.getLatestSchema(schema, false, null);
```

Comment:
The last argument is passed as "null", will that not cause NullPointer ?

Reply:
I have added an extra check within getLatestSchema (&& converterFn != null)